modules/calc/src/main/java/com/opengamma/strata/calc/Results.java

```diff
@@ -242,8 +263,124 @@ private String invalidColumnNameMessage(ColumnName columnName) {
     return cast(get(rowIndex, columnName), type);
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Returns a stream of results for a single column by column index.
+   * <p>
+   * The result may be a single value or a multi-scenario value.
+   * A multi-scenario value will implement {@link ScenarioArray} unless it has been aggregated.
+   * <p>
+   * If the calculation did not complete successfully, a failure result will be returned
+   * explaining the problem. Callers must check whether the result is a success or failure
+   * before examining the result value.
+   * <p>
+   * Large streams can be processed in parallel via {@link Stream#parallel()}.
+   * See {@link Guavate#zipWithIndex(Stream)} if the row index is required.
+   *
+   * @param <T>  the result type
+   * @param columnIndex  the index of the column
+   * @return the stream of results for the specified column
+   * @throws IllegalArgumentException if the column index is invalid
+   */
+  public <T> Stream<Result<?>> columnResults(int columnIndex) {
```

Comment:
I had one and then deleted it, providing the `ColumnName` -> index method instead. Partly because I think there may be other stream returning methods at some point, and I don't want an overload every time.

Reply:
Makes sense to me